src/server/cluster/cluster_family.cc

```diff
@@ -452,7 +452,7 @@ void ClusterFamily::DflyClusterConfig(CmdArgList args, ConnectionContext* cntx)
 
   auto cb = [&](util::ProactorBase* pb) {
     if (tl_cluster_config == nullptr) {
-      tl_cluster_config = make_unique<ClusterConfig>(*new_config);
+      tl_cluster_config = new_config;
     } else {
       *tl_cluster_config = *new_config;
```

Comment:
I think you should use tl_cluster_config.reset instead
In the current solution you update the data on all threads, this is not needed and this also might cause a bug while updating data while other thread reading the data

Reply:
Sorry, I'm not following..
`shared_ptr::reset()` takes ownership over a pointer. Since `tl_cluster_config` is thread-local, we can't have multiple `shared_ptr`s owning the same pointer as that would mean many objects will attempt to `delete` that pointer.
Further, since `tl_cluster_config` is thread-local, the update is done per-thread, so it can't be that other threads read the same data while we update it.
What am I missing?

Reply:
Also adding: generally, `shared_ptr` and `unique_ptr`'s `reset()` (and also the constructor that accepts a raw pointer) are dangerous and should be avoided.

Reply:
in line 455 you set the pointer on the first config to the shared_ptr created on calling CreateFromConfig, the shared pointer is set in all threads
Then in link 457 you get when you get an there is an update of config, what I would expect here is to have all the pointers updated and not overriding the data of ClusterConfig pointed by the shared pointer, you overide the data in all theads

Reply:
That is a **fantastic** catch! And in fact, one that I've noticed happening today, and tried to debug it briefly but to no avail. So THANK YOU!
I just sent https://github.com/dragonflydb/dragonfly/pull/1428 to fix that.